app/src/main/java/org/mozilla/focus/settings/ValidateSearchEngineAsyncTask.java

```diff
@@ -0,0 +1,130 @@
+package org.mozilla.focus.settings;
+
+
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.support.annotation.VisibleForTesting;
+import android.support.annotation.WorkerThread;
+import android.support.design.widget.Snackbar;
+import android.util.Log;
+
+import org.mozilla.focus.R;
+import org.mozilla.focus.search.ManualAddSearchEnginePreference;
+import org.mozilla.focus.search.SearchEngineManager;
+import org.mozilla.focus.telemetry.TelemetryWrapper;
+import org.mozilla.focus.utils.UrlUtils;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+
+
+
+public class ValidateSearchEngineAsyncTask extends AsyncTask<Void, Void, Boolean> {
```

Comment:
I don't think it's worth removing this from `Manual...Fragment` – we had to change the visibility of two mutable variables, which is already a really fragile pattern, so I don't want to expose them to other classes. The first alternative I think of, to use a listener to change the values, is a lot of boilerplate. I would prefer to just leave this as a private inner class of `Manual...Fragment`.

It's true that Manual... becomes a really long file and that this validation logic is a separate responsibility from managing the settings UI (so it does make sense to separate) but I think it'd be better to try to fix it by:
- Using more expressive constructs than AsyncTasks, e.g. kotlin coroutines.
- Taking *all* of the UI logic away from the validate async task
- Separating the validation logic from the async-handling logic as much as possible

Note that this async task was written last minute so it is a bit of a mess. :\